server/file-etl/run.sh

```
@@ -1,26 +1,106 @@
+    echo "✗ FILE_ETL_FROM_DB=1 needs DATABASE_URL — it reads the loaded database, not a backup." >&2
+    exit 1
+  fi
+  if [ "${FILE_ETL_APPLY:-}" = "1" ]; then
+    require_apply_env
+    echo "==> migrate-files.mjs --from-db — UPLOADING to R2 and publishing to file_url_map"
+    # The resume log seeds from the file_url_map table, not just $MAP — which matters
+    # more here than in the bulk pass, because by cutover that table is already
+    # populated and /tmp is empty. Without the table, every file the bulk pass moved
+    # would be re-uploaded under a fresh random key, orphaning the originals and
+    # invalidating the target_urls already published.
+    exec node scripts/migration/migrate-files.mjs --from-db --map="$MAP" $LIMIT_ARG
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)